app/lib/clients/vault/policy.rb

```diff
@@ -1,43 +1,81 @@
 module Clients
   class Vault
     module Policy
-      extend Entity
+      extend Identity
+      extend Oidc
 
       def rotate_token
         create_astral_policy
         token = create_astral_token
         Clients::Vault.token = token
       end
 
-      def assign_policy(identity, policy_name)
+      def assign_entity_policy(identity, policy_name)
         sub = identity.sub
         email = identity.email
         Domain.with_advisory_lock(sub) do
-          policies, metadata = get_entity_data(sub)
-          policies.append(policy_name).uniq!
-          put_entity(sub, policies, metadata)
+          put_entity(sub, [ policy_name ])
           put_entity_alias(sub, email, "oidc")
         end
       end
 
-      def verify_policy(identity, policy_name)
+      def assign_groups_policy(groups, policy_name)
+        groups.each do |group|
+          put_group(group, [ policy_name ])
+          put_group_alias(group, "#{group}-alias", "oidc")
```

Comment:
I'm pretty sure that the group_alias has to be the group name for this to work with the oidc provider.